specifyweb/frontend/js_src/lib/components/WbPlanView/Wrapped.tsx

```diff
@@ -113,6 +115,28 @@ export function WbPlanView({
     )
   );
   useErrorContext('state', state);
+  const [hasDeletedLines, setHasDeletedLines] = React.useState(false);
+
+  const handleDeleteLine = (
```

Comment:
Once you remove `hasDeletedLines`, this entire function won't be needed anymore.

This is a bit confusing but the value of `lines` and `changesMade` between Wrapped and Mapper components may differ. Wrapped has those values initially and takes care of displaying base table selection or Mapper. Then when Mapper is rendered, it uses those as default values, but doesn't need to keep Wrapped informed of updates beyond that point up until onSaved callback.

Ideally, there won't be state duplication like this, but that's how it is (this code was refactored a lot to simplify it, because previous implementation DID have one big state, but that was even worse).

Reply:
if I remove handleDeleteLine in Mapper the <Ul> is not updated. How would you do?

Reply:
Are you sure? Can you double check?
I commented out this line, and the UI was still updated properly:
https://github.com/specify/specify7/blob/beb37fef591196b658459c6e9264e00516a48814/specifyweb/frontend/js_src/lib/components/WbPlanView/Mapper.tsx#L293